### ISSUE TITLE
Phases

### ISSUE THREAD

**Issue by nicolodavis** [google/boardgame.io#51]
A declarative way to define different phases in a game:
- [x] Define which moves are permitted in the current phase.
- [x] Provide a function to customize how the turn gets passed around between players in each phase.
- [x] Option to specify if a phase is asynchronous (players may play out of turn).
- [x] Allow certain moves to happen automatically in a phase (executed by the board).
- [x] Option to end phase when all players have passed.
- [x] Add `endTurnIf` hook to allow a turn to automatically end when a condition is met.
- [x] Rename `boardgame.io/game` to `boardgame.io/core`, and expose `TurnOrder` etc. through it.
- [x] Remove `init` event and make it a `Flow` option.
- [x] Update documentation.

* *Code is being developed in the [phases](https://github.com/google/boardgame.io/tree/phases) branch.*
* *Checked items are completed.*

**Comment by nicolodavis** [comment]
Merged into `master` and released in v0.16.

**Comment by nicolodavis** [comment]
A phase is associated with the game state, not a player.

You can adjust the phase to any value as the turn gets passed around, so both option 1 and option 2 are possible.

**Comment by nicolodavis** [comment]
I'll clarify this in the examples. I'm still making some minor changes to the API, so some things are in flux.

**Comment by michaelmior** [comment]
Got it. Eventually, it would be helpful if there were a couple examples of games using phases to work from. The concepts are pretty simple, but right now I find the documentation somewhat hard to follow and I'm currently not motivated enough by this project to start digging through code. Either way, thanks for continuing to make improvements :)
